src/lib/hasFieldSelection.ts

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable require-yield */
+
+import {
+  FieldNode,
+  visit,
+  FragmentDefinitionNode,
+  BREAK,
+  GraphQLResolveInfo,
+} from "graphql"
+
+export const hasFieldSelection = (
+  resolveInfo: GraphQLResolveInfo,
+  match: (nodeName: string) => boolean
+): boolean => {
+  if (!resolveInfo.fieldNodes) return false
+  const firstFieldNode = resolveInfo.fieldNodes[0]
+
+  let matched: boolean = false
+
+  const visitor = (
+    fieldNode: FieldNode | FragmentDefinitionNode,
+    fragments?: { [key: string]: FragmentDefinitionNode }
+  ): void => {
+    visit(fieldNode, {
+      Field(node, _key, _parent, path, _ancestors) {
+        if (path.length === 3 && match(node.name.value)) {
```

Comment:
It looks like you’re now still recursing to deeper levels, I would expect something like this instead:

```suggestion
        if (path.length > 3) {
          return false
        }
        if (match(node.name.value)) {
```

Also, did you check if the `parent` parameter provides a saner way to determine depth?

Reply:
ended up addressing this as part of last commit 👍